airflow/models/baseoperator.py

```diff
@@ -1064,11 +1067,18 @@ def _set_relatives(self,
                        task_or_task_list: Union['BaseOperator', List['BaseOperator']],
                        upstream: bool = False) -> None:
         """Sets relatives for the task or task list."""
+        from airflow.models.xcom_arg import XComArg
+
         try:
             task_list = list(task_or_task_list)  # type: ignore
         except TypeError:
             task_list = [task_or_task_list]  # type: ignore
 
+        task_list = [
+            t._operator if isinstance(t, XComArg) else t  # pylint: disable=protected-access  type: ignore
+            for t in task_list
+        ]
```

Comment:
Not sure if we should allow this. Better to catch it before by making XComArg catch those.

Reply:
I am not sure if we can do this earlier to assure that both
```
XComArg >> BaseOprator
BaseOprator  >> XComArg
```
(and <<) will work

Reply:
I'm still not sure why do we need to set `>>` for XComArg. XComArg is a data unit not a compute unit. Not sure if it makes sense to specify order with those. 

I would say probs not implement this and do  https://github.com/apache/airflow/issues/8054 instead?

Reply:
Let's consider an example (not purely functional):
```python
@task 
def process_data():
    # do something amazing
    return result

with DAG("example_dag"):
    bop = BashOperator(task_id="bop", bash_command="curl ...")
    top = process_data()
 
    bop >> top 
```

Of course as you suggested this can be achieved in more functional way:
```python
@task 
def process_data(data):
    # do something amazing
    return result

with DAG("example_dag"):
    bop = BashOperator(task_id="bop", bash_command="curl ...")
    top = process_data(bop)
```

The problem probably starts once we start using sensors with functional ops:
```python
with DAG("example_dag"):
    # do something long running
    sensor = WaitForMagicSensor(task_id="sensor", ...)
    top = process_data()
 
    sensor >> top 
```